fix: breaking-changes workflow missing permissions to post PR comments

## Bug Description

The `breaking-changes.yml` workflow fails with a permissions error when trying to post a comment to PRs:

```
##[error]Unhandled error: HttpError: Resource not accessible by integration
```

The workflow successfully detects breaking changes and determines whether documentation exists, but crashes when attempting to post results as a PR comment.

## Steps to Reproduce

1. Create a PR with changes that trigger breaking change detection
2. The workflow runs and detects changes correctly
3. Workflow fails when trying to post comment via `github.rest.issues.createComment()`

## Expected vs Actual Behavior

**Expected:** Workflow posts comment to PR and completes successfully

**Actual:** Workflow crashes with 403 error:
```
RequestError [HttpError]: Resource not accessible by integration
url: 'https://api.github.com/repos/.../issues/.../comments'
status: 403
x-accepted-github-permissions: 'issues=write; pull_requests=write'
```

## Proposed Solution

Add permissions to the workflow file:

```yaml
permissions:
  contents: read
  issues: write
  pull-requests: write
```

## References

- Encountered in: https://github.com/endavis/infrafoundry/pull/145
- Failed run: https://github.com/endavis/infrafoundry/actions/runs/21260235051

--- a/.github/workflows/breaking-change-detection.yml
+++ b/.github/workflows/breaking-change-detection.yml
@@ -4,6 +4,11 @@ on:
   pull_request:
     types: [opened, synchronize, edited]
 
+permissions:
+  contents: read
+  issues: write
+  pull-requests: write
+
 jobs:
   detect-breaking-changes:
     name: Detect API Breaking Changes

